exporter/prometheusexporter/end_to_end_test.go

```diff
@@ -41,20 +42,20 @@ func TestEndToEndSummarySupport(t *testing.T) {
 	}
 
 	// 1. Create the Prometheus scrape endpoint.
-	waitForScrape := make(chan bool, 1)
-	shutdown := make(chan bool, 1)
+	var wg sync.WaitGroup
+	var currentScrapeIndex = 0
+	wg.Add(1) // scrape one endpoint
```

Comment:
Would doing `wg.Add(8)`, and then calling `wg.Done()` on each scrape work?  It would simplify the implementation

Reply:
@dashpole thanks. I think, doing `wg.Add(8)`, and then calling `wg.Done()` will only send response from HTTP test server to first 8 GET requests from Prometheus Receiver. If during the execution of this code, the Prometheus Receiver makes more than 8 requests (before the metrics are fetched from the exporter end points) then we would get  ` negative WaitGroup counter` error. This is the current problem with this test, that Prometheus Receiver makes more than 8 requests but the httpTest server only responds to first 8 requests. I think intention in this test (as was mentioned in a comment) is to make sure that HTTP test server is scraped at least 8 times by the Prometheus Receiver, but the way design was- HTTP test server only responded to first 8 - which created failed scrapes in Prometheus Receiver. In my opinion, scraping (guaranteed) HTTP test server once will produce same results as scraping 8 times (guaranteed), but I kept the same number (8) as the intention of existing implementation. In this revision, I am only ensuring that there 8 guarantee scrapes, and then the response the HTTP test server is not blocked if more GET requests are made to it by the Prometheus Receiver.

Reply:
Ah, makes sense.  I didn't realize that wg.Done() panics if it is called more than wg.Add(): https://go.dev/play/p/1FByzml9b2m.  TIL.  In that case, I think your implementation is good.

Reply:
I had the same question and learned the same thing! :)